### PR TITLE
Fix issue 28972: correct fibonacci function

### LINKED ISSUE #28972: Wrong decision in fibonacci function
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers

### What specific section or headline is this issue about?

Further examples

### What information was incorrect, unhelpful, or incomplete?

```
function fibonacci(num) {
  let a = 1;
  let b = 0;
  while (num >= 0) {
    [a, b] = [a + b, a];
    num--;
  }
  
  return b;
}

console.log(fibonacci(5)) // 8

function fib(n) {
  return n <= 1 ? n : fib(n - 1) + fib(n - 2);
}

console.log(fib(5)) // 5
```

### What did you expect to see?

fibonacci from 5 - expect 5

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/web_workers_api/using_web_workers`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/web_workers_api/using_web_workers/index.md
* Last commit: https://github.com/mdn/content/commit/bd021587852b0eeecd4d263bc20b54c6f1ef837b
* Document last modified: 2023-07-12T18:02:07.000Z

</details>

### PR DESCRIPTION
**Pull request by wbamberg**
Fixes https://github.com/mdn/content/issues/28972, using the implementation in the report, which AFAICT is correct. Thank you, @andrewkazan!

Also removed a (to me) weird note about setting event handler properties.

This PR also needs the corresponding change in dom-examples: https://github.com/mdn/dom-examples/pull/226 for a complete fix.

### CODE REVIEW

File: files/en-us/web/api/web_workers_api/using_web_workers/index.md [hunk 1]
```diff
@@ -694,24 +694,17 @@ Workers are mainly useful for allowing your code to perform processor-intensive
 The following JavaScript code is stored in the "fibonacci.js" file referenced by the HTML in the next section.
 
 ```js
-self.onmessage = (e) => {
-  const userNum = Number(e.data);
-  fibonacci(userNum);
+self.onmessage = function (event) {
```
Comment (Josh-Cena) [inline files/en-us/web/api/web_workers_api/using_web_workers/index.md:697]:
```suggestion
self.onmessage = (event) => {
```

File: files/en-us/web/api/web_workers_api/using_web_workers/index.md [hunk 2]
```diff
@@ -694,24 +694,17 @@ Workers are mainly useful for allowing your code to perform processor-intensive
...
-  let b = 0;
-  while (num >= 0) {
-    [a, b] = [a + b, a];
-    num--;
-  }
-
-  self.postMessage(b);
+function fibonacci(n) {
+  return n <= 1 ? n : fibonacci(n - 1) + fibonacci(n - 2);
 }
```
Comment (Josh-Cena) [inline files/en-us/web/api/web_workers_api/using_web_workers/index.md:704]: Let's not use recursion, especially an unoptimized one.

```js
function fibonacci(num) {
  let a = 1;
  let b = 0;
  while (num > 0) {
    [a, b] = [a + b, a];
    num--;
  }

  return b;
}
```

File: files/en-us/web/api/web_workers_api/using_web_workers/index.md [hunk 3]
```diff
@@ -736,9 +736,10 @@ The worker sets the property `onmessage` to a function which will receive messag
     <form>
       <div>
         <label for="number"
-          >Enter a number that is an index position in the fibonacci sequence to
-          see what number is in that position (e.g. enter 5 and you'll get a
-          result of 8 — fibonacci index position 5 is 8).</label
+          >Enter a number that is a zero-based index position in the fibonacci
```
Comment (Josh-Cena) [inline files/en-us/web/api/web_workers_api/using_web_workers/index.md:739]: Is this zero-based? I think Fibonacci's customarily start with 1, so 1, 1, 2, 3, 5, 8—8 is the 6th with 1-based index.

Comment (wbamberg) [reply files/en-us/web/api/web_workers_api/using_web_workers/index.md:739]: I suppose whether the argument is 0 or 1 based is up to the application to decide. Whether the sequence should start at 0 or 1 is not that clear to me:

> [The sequence commonly starts from 0 and 1, although some authors start the sequence from 1 and 1 or sometimes (as did Fibonacci) from 1 and 2.](https://en.wikipedia.org/wiki/Fibonacci_sequence)

If it starts at 1, and the index is 1-based, then the original code is correct, right:

```
| 0 | 1 | 2 | 3 | 4 | 5 |
| 1 | 1 | 2 | 3 | 5 | 8 | 
```

Honestly, I don't mind one bit. Feel free to close this PR and the issue too if you prefer.

**Review by Josh-Cena — APPROVED** [review]
Okay, looks like Fibonacci's starts with 0 outside of combinatorics. This LGTM, and adding "zero-based" surely clears up things a lot. Thanks!

**Pull request merged**